File: leetcode/linked_lists/merge_two_sorted_lists/python/test_solution.py
import unittest

from solution import solve


class MergeTwoSortedListsTests(unittest.TestCase):
    def test_examples(self) -> None:
        self.assertEqual(solve([1, 2, 4], [1, 3, 4]), [1, 1, 2, 3, 4, 4])
        self.assertEqual(solve([], [0]), [0])
        self.assertEqual(solve([1, 2, 3], []), [1, 2, 3])


if __name__ == "__main__":
    unittest.main()

File: leetcode/linked_lists/merge_two_sorted_lists/js/test_solution.js
const assert = require('assert');
const { solve } = require('./solution');

assert.deepStrictEqual(solve([1, 2, 4], [1, 3, 4]), [1, 1, 2, 3, 4, 4]);
assert.deepStrictEqual(solve([], [0]), [0]);
assert.deepStrictEqual(solve([1, 2, 3], []), [1, 2, 3]);
console.log('OK');
